manifests/osfamily/aix.pp

```diff
@@ -41,21 +43,28 @@
       $aix_ver_number = '7.1'
     } else {
       # 6.19.1 is the last puppet6 release that ships AIX 6.1 packages
+
       $aix_ver_number = versioncmp($puppet_agent::prepare::package_version, '6.19.1') ? {
         1       => '7.1',
         default => '6.1'
       }
     }
   }
 
+  if (versioncmp($pe_server_version, '2021.7.7') < 0) {
```

Comment:
@joshcooper, does this condition looks valid for backward compatibility?

Reply:
The `< 0` part looks correct:

```
❯ bx puppet apply -e '$pe_server_version = "2021.7.6"; notice(versioncmp($pe_server_version, "2021.7.7"))'
Notice: Scope(Class[main]): -1

❯ bx puppet apply -e '$pe_server_version = "2021.7.7"; notice(versioncmp($pe_server_version, "2021.7.7"))'
Notice: Scope(Class[main]): 0

❯ bx puppet apply -e '$pe_server_version = "2021.7.8"; notice(versioncmp($pe_server_version, "2021.7.7"))'
Notice: Scope(Class[main]): 1
```

But we have to support both 2021.7.x and 2023.x and we don't want to conflate the two streams. So I think you'll need to do something like:

```
if versioncmp($pe_server_version, '2021.7.7' < 0) || (versioncmp($pe_server_version, '2023.0') >= 0 && versioncmp($pe_server_version, '2023.6')) < 0) {
  # use the old way
} else {
  # use the new way
}
```

The disjoint ranges can be expressed more concisely using puppet's builtin types:

```
❯ bx puppet apply -e '$t = SemVer[SemVerRange(">= 2021.7.7 < 2023.0.0"), SemVerRange(">= 2023.6.0")]; notice(SemVer("2021.7.1") =~ $t)'
Notice: Scope(Class[main]): false

❯ bx puppet apply -e '$t = SemVer[SemVerRange(">= 2021.7.7 < 2023.0.0"), SemVerRange(">= 2023.6.0")]; notice(SemVer("2021.7.7") =~ $t)'
Notice: Scope(Class[main]): true

❯ bx puppet apply -e '$t = SemVer[SemVerRange(">= 2021.7.7 < 2023.0.0"), SemVerRange(">= 2023.6.0")]; notice(SemVer("2021.7.19") =~ $t)'
Notice: Scope(Class[main]): true

❯ bx puppet apply -e '$t = SemVer[SemVerRange(">= 2021.7.7 < 2023.0.0"), SemVerRange(">= 2023.6.0")]; notice(SemVer("2023.0.0") =~ $t)'
Notice: Scope(Class[main]): false

❯ bx puppet apply -e '$t = SemVer[SemVerRange(">= 2021.7.7 < 2023.0.0"), SemVerRange(">= 2023.6.0")]; notice(SemVer("2023.5.0") =~ $t)'
Notice: Scope(Class[main]): false

❯ bx puppet apply -e '$t = SemVer[SemVerRange(">= 2021.7.7 < 2023.0.0"), SemVerRange(">= 2023.6.0")]; notice(SemVer("2023.6.0") =~ $t)'
Notice: Scope(Class[main]): true
```

But I think there's a bug that `SemVer` requires all three components `x.y.z`. So if `pe_server_version` was just `2023.7` then it wouldn't work. Also not sure if `SemVer` handles all of the prerelease, `git describe -s`, etc in the same way as `versioncmp`?